zebra-chain/src/transaction.rs

```diff
@@ -17,7 +18,14 @@ pub use shielded_data::{Output, ShieldedData, Spend};
 pub use transparent::{CoinbaseData, OutPoint, TransparentInput, TransparentOutput};
 
 use crate::proofs::{Bctv14Proof, Groth16Proof};
-use crate::types::{amount::Amount, BlockHeight, LockTime};
+use crate::{
+    types::{amount::Amount, BlockHeight, LockTime},
+    Network,
+};
+use blake2b_simd::Hash;
+
+const OVERWINTER_VERSION_GROUP_ID: u32 = 0x03C4_8270;
+const SAPLING_VERSION_GROUP_ID: u32 = 0x892F_2085;
```

Comment:
Question: PR #856 adds a consensus parameters module to zebra-chain.
Should we keep all network upgrade constants in that module?

Question: Should these constants be a newtype, like ConsensusBranchId?

Reply:
this sounds like a great idea

Reply:
Does #856 or any other changes make these public? Previously we have been treating these as an internal implementation detail for serialization.  If we can continue doing that I don't think that having a newtype is necessary, since we'd never expose them to any other part of the code.

Reply:
I have checked the spec and ZIPs, and the Version Group Id is only used for sighash, and the transaction version checks.

So unless we want to reject transactions during parsing, we can keep it as an internal implementation detail.